service/worker/scheduler/workflow_test.go

```diff
@@ -1418,36 +1420,48 @@ func (s *workflowSuite) TestLotsOfIterations() {
 		}
 	}
 
+	delayedCallbacks := make([]delayedCallback, backfillIterations)
+
+	expected := runIterations
+	// schedule a call back every hour to spray backfills among scheduled runs
```

Comment:
one nitpick: can you make the backfill only start after 2022-06-01 15:00:00? i.e. I want at least one long stretch where the cache runs out and has to refill naturally. (or even better, use the value of `maxNextTimeResultCacheSize`.) I think it may not actually be covering that branch anymore